source/termination_criteria/steady_heat_flux.cc

```diff
@@ -0,0 +1,212 @@
+        // but ensure at least two entries remain in the list (one old, one current timestep)
+        auto it = time_heat_flux_list.begin();
+        while (time_heat_flux_list.back().first - (*it).first > necessary_time_in_steady_state &&
+               std::distance(it,time_heat_flux_list.end()) > 2)
+          ++it;
+
+        time_heat_flux_list.erase(time_heat_flux_list.begin(), it);
+      }
+    }
+
+    template <int dim>
+    bool
+    SteadyHeatFlux<dim>::execute()
+    {
+      std::vector<std::vector<std::pair<double, double> > > heat_flux_and_area =
+        Postprocess::internal::compute_heat_flux_through_boundary_faces (*this);
+
+      double local_boundary_fluxes = 0.0;
+
+      for (const auto &cell : this->get_dof_handler().active_cell_iterators())
+        if (cell->is_locally_owned())
+          for (unsigned int f=0; f<GeometryInfo<dim>::faces_per_cell; ++f)
+            if (cell->at_boundary(f))
+              {
+                const types::boundary_id boundary_indicator
+                  = cell->face(f)->boundary_id();
+                if (boundary_indicators.find(boundary_indicator) != boundary_indicators.end())
+                  local_boundary_fluxes += heat_flux_and_area[cell->active_cell_index()][f].first;
+              }
+
+      const double global_heat_flux_integral
+        = Utilities::MPI::sum (local_boundary_fluxes, this->get_mpi_communicator());
+
+      // Keep a list of times and heat fluxes at those times
+      time_heat_flux.emplace_back(this->get_time(), global_heat_flux_integral);
+
+      // If the length of the simulation time covered in the list is shorter than the
+      // specified parameter, we must continue the simulation
+      if ((time_heat_flux.size() <= 2)
+          ||
+          (time_heat_flux.back().first - time_heat_flux.front().first < necessary_time_in_steady_state))
+        return false;
+
+      // Remove old entries outside of current time window
+      internal::trim_time_heat_flux_list(necessary_time_in_steady_state,time_heat_flux);
+
+      // Scan through the list and calculate the min, mean and max heat flux
+      // We assume a linear change of heat flux between times
+      double flux_min, flux_max, flux_prev, time_prev, flux_sum=0, flux_mean, deviation_max;
+      flux_min = flux_max = flux_prev = time_heat_flux.front().second;
+      time_prev = time_heat_flux.front().first;
+      for (auto it=time_heat_flux.begin(); it!=time_heat_flux.end(); ++it)
+        {
+          flux_min = std::min(flux_min, (*it).second);
+          flux_max = std::max(flux_max, (*it).second);
+          flux_sum += (((*it).second + flux_prev)/2.0)*((*it).first-time_prev);
+          time_prev = (*it).first;
+          flux_prev = (*it).second;
+        }
+
+      flux_mean = flux_sum/(time_heat_flux.back().first-time_heat_flux.front().first);
+
+      // If the min and max are within the acceptable deviation of the mean,
```

Comment:
A general question about the approximations being made here: if `time_heat_flux` only contains two entries, that would mean the steady state heat flux is being computed from adjacent time steps. 

Is this reasonable? Or would this only happen on the second time step? 

I'm trying to remember how I used to determine when steady was reached and I recall using quite a few time steps. Again, likely a mute point.

Reply:
Yes, that's right. But I would say, in that case, the user should pick a different "Time in steady state" in the input file, or set a minimum time step that is smaller than the "Time in steady state". If the time steps are substantially smaller than that, there should always be enough entries in the `time_heat_flux` vector.

Reply:
@jdannberg - Makes sense, thanks for the explanation! Could you add some of the above text either at the top of the test case prm file and /or in the documentation for the "Time in steady state" parameter? You could almost copy this verbatim into the prm file. Other than that, ready to be merged!

Reply:
I have added a few lines to the documentation of the input parameter. I thought that would be the place where it is most useful for the user.